pkg/tkctl/cmd/check/check.go

```diff
@@ -0,0 +1,460 @@
+			}
+			infoPrintf("TiKV rocksdb.lockcf.block-cache-size: %dMiB\n", tikvcfg.Rocksdb.Lockcf.BlockCacheSize/1024/1024)
+		}
+		if !tikvcfg.RaftStore.SyncLog {
+			errorPrintln("Raftstore sync-log is disable")
+		}
+		if tikvcfg.LogLevel != "info" {
+			warnPrintln("TiKV log-level:", tikvcfg.LogLevel)
+		}
+	}
+	if cfg, exist := pumpConfigMap.Data["pump-config"]; exist {
+		infoPrintln("#### Pump Config ####")
+		fmt.Println(cfg)
+		infoPrintln("#### Pump Config ####\n")
+	}
+	if cfg, exist := pdConfigMap.Data["config-file"]; exist {
+		infoPrintln("#### PD Config ####")
+		fmt.Println(cfg)
+		infoPrintln("#### PD Config ####\n")
+		_, err = toml.Decode(cfg, &pdcfg)
+		if err != nil {
+			return fmt.Errorf("failed to decode pd config: %v", err)
+		}
+		if pdcfg.Log.Level != "info" {
+			warnPrintln("PD log-level:", pdcfg.Log.Level)
+		}
+		if pdcfg.Replication.MaxReplicas != 3 {
+			warnPrintln("TiKV data replication factor is", pdcfg.Replication.MaxReplicas)
+		}
+	}
+	return nil
+}
+
+func checkResourceSettings(tc *v1alpha1.TidbCluster, pump *appsv1.StatefulSet, drainer *appsv1.StatefulSet) {
+	pdResources := util.ResourceRequirement(tc.Spec.PD.ContainerSpec)
+	tikvResources := util.ResourceRequirement(tc.Spec.TiKV.ContainerSpec)
+	tidbResources := util.ResourceRequirement(tc.Spec.TiDB.ContainerSpec)
+	if err := compareResources(pdResources); err != nil {
+		errorPrintf("PD resources settings (%v) error: %v\n", pdResources, err)
+	}
+	if err := compareResources(tikvResources); err != nil {
+		errorPrintf("TiKV resources settings (%v) error: %v\n", tikvResources, err)
+	}
+	if err := compareResources(tidbResources); err != nil {
+		errorPrintf("TiDB resources settings (%v) error: %v\n", tidbResources, err)
+	}
+
+	if container := getContainerFromSts(pump, "pump"); container != nil {
+		if err := compareResources(container.Resources); err != nil {
+			errorPrintf("Pump resources settings (%v) error: %v\n", container.Resources, err)
+		}
+	}
+	if container := getContainerFromSts(drainer, "drainer"); container != nil {
+		if err := compareResources(container.Resources); err != nil {
+			errorPrintf("Drainer resources settings (%v) error: %v\n", container.Resources, err)
+		}
+	}
+}
+
+// check requests and limits cpu/memory not zero, requests <= limits, (warn if cpu is not integer, or not Guaranteed QoS)
+func compareResources(requirement corev1.ResourceRequirements) error {
+	cpuRequest := requirement.Requests[corev1.ResourceCPU]
+	cpuLimit := requirement.Limits[corev1.ResourceCPU]
+	memoryRequest := requirement.Requests[corev1.ResourceMemory]
+	memoryLimit := requirement.Limits[corev1.ResourceMemory]
+
+	if cpuRequest.IsZero() || cpuLimit.IsZero() {
+		return fmt.Errorf("cpu request or limit is zero")
+	}
+	switch cpuRequest.Cmp(cpuLimit) {
+	case -1:
+		warnPrintln("CPU request < CPU limit, not Guaranteed QoS")
```

Comment:
Messages printed in this method should print which component fail the check.

Alternatively we can print a message when entering this method to declare which component are being checked.